src/NimBLEUUID.cpp

```diff
@@ -42,50 +44,25 @@ static const char* LOG_TAG = "NimBLEUUID";
     m_valueSet = true;
     if (value.length() == 4) {
```

Comment:
I think this should be changed to use the NimBLE function so the byte order is correct.
Possibly this:
```
if (value.length() == 4 || value.length() == 8 ) {
    uint32_t buf = strtoul(value.c_str(), NULL, 16);
    int rc = ble_uuid_init_from_buf(m_uuid, buf, value.length());
    if(rc != 0) {
        NIMBLE_LOGE(LOG_TAG, "Invalid UUID");
        return;
    }
}
```
That would eliminate the first 2 conditionals.

Untested but should work.

Reply:
Nevermind, wasn't thinking, this isn't necessary.